app/src/main/java/com/github/siela1915/bootcamp/MainHomeActivity.java

```diff
@@ -217,36 +216,59 @@ private void loadAndOpenFavorites() {
         setContainerContent(R.id.fragContainer, RecipeListFragment.newInstance(new ArrayList<>()), false);
 
         Database db = new Database(FirebaseDatabase.getInstance());
-        db.getFavorites().continueWith(favTask -> {
-            List<String> favorites = favTask.getResult();
+        db.getFavorites().addOnSuccessListener(favorites -> {
             List<Task<Recipe>> favListTasks = favorites.stream().map(db::getAsync).collect(Collectors.toList());
             Tasks.whenAll(favListTasks).addOnSuccessListener(voidRes -> {
                 Fragment currentFrag = fragmentManager.findFragmentById(R.id.fragContainer);
                 if (currentFrag == null || currentFrag.getClass() != RecipeListFragment.class) {
                     return;
                 }
+                fragmentManager.popBackStack();
                 List<Recipe> favRecipes = favListTasks
                         .stream().map(Task::getResult).collect(Collectors.toList());
                 setContainerContent(R.id.fragContainer, RecipeListFragment.newInstance(
                         favRecipes
                 ), false);
             });
-            return null;
         });
     }
 
+    private void openHelp() {
+        Bundle extras = getIntent().getExtras();
+        navigationView.setCheckedItem(R.id.menuItem_help);
+        Intent intent = new Intent(this, MainHomeActivity.class);
+        if (extras != null) {
+            intent.putExtras(extras);
+        }
+        intent.putExtra("navToHelp", true);
+        FirebaseAuthActivity.promptLogin(this, this, intent);
+
+        if (extras != null && extras.containsKey("sender")) {
+            setContainerContent(R.id.fragContainer, NearbyHelpFragment.newInstance(
+                            extras.getString("sender"),
+                            extras.getString("ingredient")),
+                    false);
+        } else if (extras != null && extras.containsKey("askedIngredient")) {
+            setContainerContent(R.id.fragContainer, NearbyHelpFragment.newInstance(
+                            extras.getParcelable("askedIngredient")),
```

Comment:
Same comment here. Storing those strings in variables should be the future improvement. :)